tests/output/test_io_timestepping.py

```diff
@@ -74,3 +75,42 @@ def test_io_timestepping(element, tmpdir):
                 g = Function(V)
                 _project(g, _get_expr(V, i), method)
                 assert assemble(inner(g - f, g - f) * dx) < 1.e-16
+
+
+def test_io_timestepping_setting_time(tmpdir):
+    filename = os.path.join(
+        str(tmpdir), "test_io_timestepping_setting_time_dump.h5")
+    filename = COMM_WORLD.bcast(filename, root=0)
+    mesh = UnitSquareMesh(5, 5)
+    RT2_space = VectorFunctionSpace(mesh, "RT", 2)
+    cg1_space = FunctionSpace(mesh, "CG", 1)
+    mixed_space = MixedFunctionSpace([RT2_space, cg1_space])
+    z = Function(mixed_space, name="z")
+    u, v = z.subfunctions
+    u.rename("u")
+    v.rename("v")
+
+    indices = range(0, 10, 2)
+    ts = np.linspace(0, 1.0, len(indices))*2*np.pi
+    timesteps = np.linspace(0, 1.0, len(indices))
+
+    with CheckpointFile(filename, mode="w") as f:
+        f.save_mesh(mesh)
+        for idx, t, timestep in zip(indices, ts, timesteps):
+            u.assign(t)
+            v.interpolate((cos(Constant(t)/pi)))
+            f.save_function(z, idx=idx, timestepping_info={"time": t, "timestep": timestep})
+
+    with CheckpointFile(filename, mode="r") as f:
+        mesh = f.load_mesh(name="firedrake_default")
+        timestepping_history = f.get_timestepping_history(mesh, name="u")
+        loaded_v = f.load_function(mesh, "v", idx=timestepping_history.get("indices")[-2])
+
+    assert (indices == timestepping_history.get("indices")).all()
+    assert (ts == timestepping_history.get("time")).all()
+    assert (timesteps == timestepping_history.get("timestep")).all()
+
+    # checking if the function is exactly what we think
+    v_answer = interpolate(cos(Constant(timestepping_history.get("time")[-2])/pi),
+                           loaded_v.function_space())
```

Comment:
```suggestion
    v_answer = Function(loaded_v.function_space()).interpolate(cos(Constant(timestepping_history.get("time")[-2])/pi))
```
How `interpolate` function works will change in the near future. This change will avoid warnings after you update your branch.